istioctl/pkg/multixds/gather.go

```diff
@@ -121,7 +121,9 @@ func queryEachShard(all bool, dr *discovery.DiscoveryRequest, istioNamespace str
 			return nil, fmt.Errorf("could not get XDS from discovery pod %q: %v", pod.Name, err)
 		}
 
-		if proxyNotConnectedToThisPilotInstanceResponse(response) {
+		if idx < len(pods)-1 && proxyNotConnectedToThisPilotInstanceResponse(response) {
```

Comment:
It is confusing to me without reading the whole function

Reply:
Agreed, updated comments to help understanding。